File: SDKs/amazonaws.com/application-insights/2018-11-25/go/pkg/models/operations/listcomponents.go
package operations

import (
	"openapi/pkg/models/shared"
)

type ListComponentsQueryParams struct {
	MaxResults *string `queryParam:"style=form,explode=true,name=MaxResults"`
	NextToken  *string `queryParam:"style=form,explode=true,name=NextToken"`
}

type ListComponentsXAmzTargetEnum string

const (
	ListComponentsXAmzTargetEnumEc2WindowsBarleyServiceListComponents ListComponentsXAmzTargetEnum = "EC2WindowsBarleyService.ListComponents"
)

type ListComponentsHeaders struct {
	XAmzAlgorithm     *string                      `header:"style=simple,explode=false,name=X-Amz-Algorithm"`
	XAmzContentSha256 *string                      `header:"style=simple,explode=false,name=X-Amz-Content-Sha256"`
	XAmzCredential    *string                      `header:"style=simple,explode=false,name=X-Amz-Credential"`
	XAmzDate          *string                      `header:"style=simple,explode=false,name=X-Amz-Date"`
	XAmzSecurityToken *string                      `header:"style=simple,explode=false,name=X-Amz-Security-Token"`
	XAmzSignature     *string                      `header:"style=simple,explode=false,name=X-Amz-Signature"`
	XAmzSignedHeaders *string                      `header:"style=simple,explode=false,name=X-Amz-SignedHeaders"`
	XAmzTarget        ListComponentsXAmzTargetEnum `header:"style=simple,explode=false,name=X-Amz-Target"`
}

type ListComponentsRequest struct {
	QueryParams ListComponentsQueryParams
	Headers     ListComponentsHeaders
	Request     shared.ListComponentsRequest `request:"mediaType=application/json"`
}

type ListComponentsResponse struct {
	ContentType               string
	InternalServerException   *interface{}
	ListComponentsResponse    *shared.ListComponentsResponse
	ResourceNotFoundException *interface{}
	StatusCode                int64
	ValidationException       *interface{}
}

File: SDKs/amazonaws.com/application-insights/2018-11-25/typescript/src/sdk/models/operations/listcomponents.ts
import { Metadata, SpeakeasyBase } from "../../../internal/utils/utils";
import * as shared from "../shared";


export class ListComponentsQueryParams extends SpeakeasyBase {
  @Metadata({ data: "queryParam, style=form;explode=true;name=MaxResults" })
  maxResults?: string;

  @Metadata({ data: "queryParam, style=form;explode=true;name=NextToken" })
  nextToken?: string;
}

export enum ListComponentsXAmzTargetEnum {
    Ec2WindowsBarleyServiceListComponents = "EC2WindowsBarleyService.ListComponents"
}


export class ListComponentsHeaders extends SpeakeasyBase {
  @Metadata({ data: "header, style=simple;explode=false;name=X-Amz-Algorithm" })
  xAmzAlgorithm?: string;

  @Metadata({ data: "header, style=simple;explode=false;name=X-Amz-Content-Sha256" })
  xAmzContentSha256?: string;

  @Metadata({ data: "header, style=simple;explode=false;name=X-Amz-Credential" })
  xAmzCredential?: string;

  @Metadata({ data: "header, style=simple;explode=false;name=X-Amz-Date" })
  xAmzDate?: string;

  @Metadata({ data: "header, style=simple;explode=false;name=X-Amz-Security-Token" })
  xAmzSecurityToken?: string;

  @Metadata({ data: "header, style=simple;explode=false;name=X-Amz-Signature" })
  xAmzSignature?: string;

  @Metadata({ data: "header, style=simple;explode=false;name=X-Amz-SignedHeaders" })
  xAmzSignedHeaders?: string;

  @Metadata({ data: "header, style=simple;explode=false;name=X-Amz-Target" })
  xAmzTarget: ListComponentsXAmzTargetEnum;
}


export class ListComponentsRequest extends SpeakeasyBase {
  @Metadata()
  queryParams: ListComponentsQueryParams;

  @Metadata()
  headers: ListComponentsHeaders;

  @Metadata({ data: "request, media_type=application/json" })
  request: shared.ListComponentsRequest;
}


export class ListComponentsResponse extends SpeakeasyBase {
  @Metadata()
  contentType: string;

  @Metadata()
  internalServerException?: any;

  @Metadata()
  listComponentsResponse?: shared.ListComponentsResponse;

  @Metadata()
  resourceNotFoundException?: any;

  @Metadata()
  statusCode: number;

  @Metadata()
  validationException?: any;
}
